azkaban-common/src/main/java/azkaban/project/FlowLoaderUtils.java

```diff
@@ -340,4 +343,63 @@ public boolean accept(final File pathname) {
       return pathname.isDirectory();
     }
   }
+
+  /**
+   * Loads flow/job properties from the flow's YAML file. If path is null
+   * then it loads the flow's properties, otherwise it loads the property
+   * of the job at the path. The caller is responsible for providing the
+   * correct path
+   * @param projectLoader Used to fetch file from DB.
+   * @param executableFlow The executable flow of which properties are
+   *                       being loaded.
+   * @param path Path to job file. NULL for flow properties.
+   * @return return Props object with flow/job properties.
+   */
+  public static Props loadPropsFromYamlFile(final ProjectLoader projectLoader,
+      final ExecutableFlow executableFlow, final String path) {
+    File tempDir = null;
+    Props props = null;
+    try {
+      tempDir = com.google.common.io.Files.createTempDir();
+      props = FlowLoaderUtils.getPropsFromYamlFile(
+          path == null ? executableFlow.getId() : path,
+          getFlowFile(tempDir, projectLoader, executableFlow));
+    } catch (final Exception e) {
+      logger.error("Failed to get props from flow file. " + e);
+    } finally {
+      if (tempDir != null && tempDir.exists()) {
+        try {
+          FileUtils.deleteDirectory(tempDir);
+        } catch (final IOException e) {
+          logger.error("Failed to delete temp directory." + e);
+          tempDir.deleteOnExit();
+        }
+      }
+    }
+    return props;
+  }
+
+  /**
+   * This function fetches the flow file and puts it in tempDir
+   * @param tempDir location where the flow file is put
+   * @param projectLoader Used to fetch from DB
+   * @param flow the executable flow
+   * @return returns the flow file from db.
+   * @throws Exception
+   */
+  private static File getFlowFile(final File tempDir, final ProjectLoader projectLoader,
+      final ExecutableFlow flow) throws Exception {
+    final List<FlowProps> flowPropsList = ImmutableList.copyOf(flow.getFlowProps());
```

Comment:
Can flow.getFlowProps() be null?

Reply:
No. Besides this is a refactored code lifted from FlowRunner where it has been in use for years.